Adding a QUIC section to ssl.rst (#232)
zeek/zeek-docs: logs/quic.rst

====
QUIC
====

.. versionadded:: 6.1

The QUIC protocol integrates encryption, stream multiplexing and flow control at

> the transport layer. QUIC uses TLS 1.3 by default. The QUIC spicy analyzer

Reviewer: No need to mention spicy.
```suggestion
the transport layer. QUIC uses TLS 1.3 by default. Zeek's QUIC analyzer
```

Author: ✅

attempts to handle the QUIC INITIAL packet payload to provide greater

> observability into the protocol, including the server name.

Reviewer: maybe shorten as "to provide visibility into the TLS handhake".

It does a few things more like retry packet handling, but yes, it's not a full protocol analyzer.

Author: ✅

quic.log
========

An example of a :file:`quic.log`.

.. code-block:: console

    zeek@zeek-6.1:~ zeek -C LogAscii::use_json=T -r chromium-115.0.5790.110-api-cirrus-com.pcap
    zeek@zeek-6.1:~ jq . quic.log

::

  {
    "ts": 1692198386.837988,
    "uid": "CA482y1XJVd3d0RYI7",
    "id.orig_h": "82.239.54.117",
    "id.orig_p": 53727,
    "id.resp_h": "110.213.53.115",
    "id.resp_p": 443,
    "version": "1",
    "client_initial_dcid": "95412c47018cdfe8",
    "server_scid": "d5412c47018cdfe8",
    "server_name": "api.cirrus-ci.com",
    "client_protocol": "h3",
    "history": "ISisH"
  }

QUIC Info
=========

:zeek:see:`QUIC::Info` provides further details on the current output of the
:file:`quic.log`. Current fields include:

- **version**: The version of QUIC that was identified. Note, QUICv2 has not
  been widely tested.

- **client_initial_dcid**: When QUIC initiates a connection it uses Random
  Number Generators to create the first Destination Connection ID (DCID). This
  DCID is subsequently used for routing and packet protection by client and
  server.

- **server_scid**: A QUIC supported server responds to a DCID by

>   selecting a Source Connection ID (SCID). This usually occur within the

Reviewer: ```suggestion
  selecting a Source Connection ID (SCID). This usually occurs within the
```

Author: ✅

  server’s first INITIAL packet. This is typically used by the client in
  subsequent packets, although the SCID can change to adapt to new network
  conditions.

- **client_protocol**: During the Application-Layer Protocol Negotiation
  (ALPN) extension, if the ClientHello packet is available, it is extracted.

>   In the example above, we can see that HTTPv3 is the client protocol.

Reviewer: This is a bit backwards. I'd say "If the ClientHello packet is successfully extracted and contains the ALPN extension, the extension's first entry is placed in `client_protocol`.

Author: ✅

- **history**: An experimental feature provides a history of the QUIC
  protocol usage. In the example above the history outlines:

>     + An initial packet from the client - a new connection

Reviewer: ```suggestion
    + An initial packet from the client (I) - a new connection
```

Author: ✅

>     + An SSL Client/Server Hello from the client - the start of a

Reviewer: ```suggestion
    + An SSL ClientHello from the client (S) - the start of a
```

etc.

Author: ✅

      SSL/TLS handshake

    + An initial packet response from the server - an acknowledgement
      from the server of the new connection

>     + An SSL Client/Server Hello response from the server - the

Reviewer: ```suggestion
    + An SSL ServerHello response from the server (s) - the
```

      selection  of a cipher suite from the options provided by the
      client

>     + A handshake packet - completes the TLS negotiation

Reviewer: ```suggestion
    + A handshake packet from the client (H)
```

I don't think these necessarily complete the negotiation, so would leave it out. Also might be good to avoid mixing SSL and TLS and stick with one of them in the docs.

Author: ✅

Conclusion
==========

The QUIC analyzer is an initial attempt to provide observability into QUIC
network traffic. In Zeek 6.1, although included by default, the QUIC analyzer is
still considered experimental. In addition to code contributions, feedback and

> issues are greatly appreciated, and can be reported via Zeek github, slack or

Author: ✅
